### DISCUSSION TITLE
Short Vec types

### DISCUSSION THREAD

**Discussion by anderslanglands** [bitshifter/glam-rs#260] [Ideas]
Hi, I'm a big fan of your crate for graphics stuff. One thing that's missing that would be useful for vulkan, optix etc would be a 16-bit integer vec type, basically IVecX but with i16 instead of i32 (SVec3 etc perhaps?). Would you accept a PR adding this?

**Comment by bitshifter** [comment]
Hi there!

I'm interested in hearing a bit more about your use case for `i16` vector types if you can talk about it.

This is a bit of a Pandora's Box because obviously there are a lot of types that `glam` still doesn't support and not being a generic lib each type needs to have their implementation added. That said, if this is a common thing that is needed then I am not opposed to adding them, although I have so far resisted in some cases (e.g. #232). It isn't that hard to add new types as most of the implementation could reuse the existing macros.

It seems that `i16` etc. extensions to glsl have been proposed by [nvidia](https://github.com/KhronosGroup/GLSL/blob/master/extensions/ext/GL_EXT_shader_16bit_storage.txt) and [amd](https://www.khronos.org/registry/OpenGL/extensions/AMD/AMD_gpu_shader_int16.txt). Naming wise I've tended to use glsl style naming so far so that's what I'd suggest (with Rust capitialization, e.g. `I16Vec3`).

tl;dr I would be open to a PR. It might make sense to have this on a feature toggle initially, it would be good to check if there's any noticeable compile time / size increase from expanding the number of types, especially if they expand beyond `i16`. If they're more niche a feature would mean general users wouldn't need to pay for things they don't use.

**Reply by kawogi** [reply]
In Minetest (block/voxel game) the internal coordinate system relies on `i16x3`. I just wanted to change the manual vector implementation ( https://github.com/UgnilJoZ/rust-minetestworld/blob/main/src/positions.rs ) to some `glam`-type but didn't find a suitable one. Instead I found this thread and wanted to add my use case to the list :)

Another use case I had back then was audio data, which is commonly stored and transmitted as `i16x2` (not so much being used for computations, though)

Screen-coordinates have been 16 bits wide in the past, but I think this is being phased out.

What change of acceptance would a PR have? ;)

**Comment by kawogi** [comment]
relevant PR: https://github.com/bitshifter/glam-rs/pull/439

re: compile times - a clean build took 2.2 s on my machine. I don't know the original time but in absolute measures, this look fine to me.

**Comment by bitshifter** [comment]
Merged in #439
